docs/src/DevelopersGuide.md

```diff
@@ -0,0 +1,180 @@
+  and tag the issue it is addressing.
+In the PR you can describe which parts of the relevant issue are solved and what needs to
+  be done to reach that goal.
+After creating a PR on GitHub, you will find that every following commit & push you make
+  will go through continuous integration (CI) checks:
+
+- The, `ci / ci 1.8.1 ` runs the tests on Ubuntu, Windows and OSX machines
+  using GitHub Actions.
+  The tests include some unit tests and very simple performance tests.
+  It may happen that the tests pass on your local machine,
+  but fail on one of the machines provided in the cloud for the CI.
+  This is especially true for the performance tests,
+  as the execution times may vary a lot depending on the machine.
+  See the [Tests](https://clima.github.io/CloudMicrophysics.jl/dev/DevelopersGuide/#Documentation)
+  section below for debugging tips.
+- `CloudMicrophysics.jl` has a small set of tests that are run on the GPUs using `buildkite`.
+  They are triggered automatically when trying to merge a PR,
+  or manually by commenting `bors try` on your PR.
+  If the GPU tests fail, check first for use of any "out of place" functions
+  (for example, `@warn` will not work on GPU).
+- The `Documentation / docbuild` builds the documentation.
+  Most common errors in the docbuild are related to missing docstrings,
+  see the [Documentation](https://clima.github.io/CloudMicrophysics.jl/dev/DevelopersGuide/#Documentation) section for hints.
+  This check may fail if the source code itself is not compiling,
+  so please handle the compilation errors first.
+  If the documentation build was successful, the `documenter/deploy`
+  will display the documentation page based on the PR (click on the details).
+- The `JuliaFormatter / format` ensures consistent formatting throughout the repository.
+  If this check fails, you can click on details
+  to see which files are not following the formatting rules.
+  You can apply the formatter by running
+  `julia --project=.dev .dev/climaformat.jl file_name` in the terminal.
+  The formatter might break if the code does not compile,
+  so make sure you handle the compilation errors first.
+- The `codecov` shows what percentage of source code lines are exercised inside the tests.
+  We strive to keep the test coverage high, but this test is not strictly required to pass before merging a PR.
+
+Once the PR is opened, you can request reviewers to look over and approve your work.
+To keep things tidy, we want PRs to have very few (ideally only one) commit
+  before merging to the main branch.
+You can squash and rebase multiple commits into one
+  in your favorite editor (VS Code, Vim etc).
+In case of doubt, see Git tutorials on how to squash and rebase
+  or reach out to other developers in our team.
+The first couple of times it pays off to create a "backup branch" before starting your rebase,
+  and then comparing afterwards if the rebased and backup branches are identical.
+
```

Comment:
If the main branch has been updated after your branch was created, you will need to rebase onto the main branch. To do this, run `git rebase origin/main` and solve any conflicts manually. This is done more easily if you only have one commit.